async_fuse/src/aligned_bytes.rs

```diff
@@ -105,7 +110,8 @@ mod tests {
         let align = 4096;
         let bytes = AlignedBytes::new_zeroed(8, align);
         assert_eq!(bytes.alignment(), align);
-        assert!(bytes.as_ptr() as usize % align == 0);
+        let address = unsafe { std::mem::transmute::<*const u8, usize>(bytes.as_ptr()) };
```

Comment:
According to the document, there are some better way to transfer type instead of transmute, such as "as" cast.

Reply:
We denied all clippy lint errors, and clippy has a lint error about as conversion:
https://rust-lang.github.io/rust-clippy/master/#as_conversions